Doc/library/stdtypes.rst

```diff
@@ -4749,6 +4749,198 @@ define these methods must provide them as a normal Python accessible method.
+<generic type>`, implementing *parameterized generics* - a specific instance
+of a generic with the types for container elements provided. It is intended
+primarily for type :term:`annotations <annotation>`.  ``GenericAlias`` objects
+are created by expressions like ``list[int]``.
+
+The type for the ``GenericAlias`` object is :data:`types.GenericAlias`.
+
+.. impl-detail::
+
+   The C equivalent of :meth:`__getitem__` - :c:func:`PyObject_GetItem` - is
+   called first for builtin classes and types. When :c:func:`PyObject_GetItem`
+   detects that it was called on a type, it then calls :meth:`__class_getitem__`
+   and returns the ``GenericAlias`` object.
+
+.. describe:: generic[X, Y, ...]
+
+   Defines a generic containing elements of types *X*, *Y*, and more depending
+   on the *generic* used.  For example, for a function expecting a :class:`list`
+   containing :class:`float` elements::
+
+      def average(values: list[float]) -> float:
+          return sum(values) / len(values)
+
+   Another example for :term:`mapping` objects, using a :class:`dict`.  In this
+   case, the function expects a ``dict`` for its ``body`` argument. The ``body``
+   has keys of type :class:`str` and their corresponding values are lists which
+   hold :class:`int` elements::
+
+      def send_post_request(url: str, body: dict[str, list[int]]) -> None:
+          ...
+
+The builtin functions :func:`isinstance` and :func:`issubclass` do not accept
+parameterized generic types for their second argument::
+
+   >>> isinstance([1, 2], list[str])
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   TypeError: isinstance() argument 2 cannot be a parameterized generic
+   >>>
+   >>> issubclass(list, list[str])
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   TypeError: issubclass() argument 2 cannot be a parameterized generic
+
+The Python runtime does not enforce type :term:`annotations <annotation>`.
+This extends to generic types and their type parameters. When creating
+an object from a parameterized generic, container elements are not checked
+against their type. For example, the following code is discouraged, but will
+run without errors::
+
+   >>> t = list[str]
+   >>> t([1, 2, 3])
+   [1, 2, 3]
+
+Furthermore, parameterized generics erase type parameters during object
+creation::
+
+   >>> t = list[str]
+   >>> type(t)
+   <class 'types.GenericAlias'>
+   >>>
+   >>> l = t()
+   >>> type(l)
+   <class 'list'>
+
+Calling :func:`repr` on a generic shows the parameterized type::
+
+   >>> repr(list[int])
+   'list[int]'
+
+The :meth:`__getitem__` method of generics will raise an exception to disallow
+mistakes like ``dict[str][str]``::
+
+   >>> dict[str][str]
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in <module>
+   TypeError: There are no type variables left in dict[str]
+
+However, that syntax is valid when type variables are used::
```

Comment:
Can you rewrite this sentence without referring to syntax? It's not a syntactic issue -- the rule is that a GenericAlias may only be indexed if it has type variables, and the index must have as many elements as there are items in its `__args__`.